File: javascript/2221.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');
const values = input.split(/\s+/).map((value) => parseInt(value));

function pomekonsBattle(bonus, player1PokemonsAttack, player2PokemonsAttack) {
  const [player1Ai, player1Di, player1Li] = player1PokemonsAttack;
  const [player2Ai, player2Di, player2Li] = player2PokemonsAttack;
  let player1TotalAttack = (player1Ai + player1Di) / 2.0;
  let player2TotalAttack = (player2Ai + player2Di) / 2.0;

  if (player1Li % 2 === 0) {
    player1TotalAttack += bonus;
  }

  if (player2Li % 2 === 0) {
    player2TotalAttack += bonus;
  }

  if (player1TotalAttack > player2TotalAttack) {
    return "Dabriel";
  } else if (player2TotalAttack > player1TotalAttack) {
    return "Guarte";
  }

  return "Empate!";
}

let game = values.shift();

while (game > 0) {
  const bonus = values.shift();
  const player1PokemonsAttack = values.splice(0, 3);
  const player2PokemonsAttack = values.splice(0, 3);
  console.log(
    pomekonsBattle(bonus, player1PokemonsAttack, player2PokemonsAttack)
  );
  game--;
}

File: java/2221.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
	
    public static void main(String[] args) throws IOException {
    	Scanner leitor = new Scanner(System.in);
    	int T = leitor.nextInt();
    	int B, A1, D1, L1, A2, D2, L2, V1, V2;
    	for (int i = 0; i < T; i++) {
    		B = leitor.nextInt();
    		A1 = leitor.nextInt();
    		D1 = leitor.nextInt();
    		L1 = leitor.nextInt();
    		A2 = leitor.nextInt();
    		D2 = leitor.nextInt();
    		L2 = leitor.nextInt();
    		V1 = ((A1 + D1) / 2) + (L1 % 2 == 0 ? B : 0);
    		V2 = ((A2 + D2) / 2) + (L2 % 2 == 0 ? B : 0);
    		if (V1 > V2) System.out.println("Dabriel");
    		else if (V2 > V1) System.out.println("Guarte");
    		else System.out.println("Empate");
    	}
    }
	
}
